delivery_price_rule_untaxed/models/delivery_carrier.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import safe_eval
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    def _get_price_available(self, order):
+        self.ensure_one()
+        self = self.with_context({"order_amount_untaxed": order.amount_untaxed})
+        res = super(DeliveryCarrier, self)._get_price_available(order)
+        return res
```

Comment:
```suggestion
        return super(DeliveryCarrier, **self)._get_price_available(order)
```

Reply:
I didn't see that one.

But no, the good one is :
```suggestion
        new_self = self.with_context({"order_amount_untaxed": order.amount_untaxed})
        return super(DeliveryCarrier, new_self)._get_price_available(order)
```
As you cannot replace self.

Reply:
done, thanks